packages/react-aria-components/src/Menu.tsx

```diff
@@ -100,8 +139,18 @@ function MenuInner<T extends object>({props, collection, menuRef: ref}: MenuInne
     collection,
     children: undefined
   });
+  let rootMenuTriggerState = useContext(RootMenuTriggerStateContext)!;
+  let popoverContainerRef = useRef<HTMLDivElement>(null);
   let {menuProps} = useMenu(props, state, ref);
 
+  let [leftOffset, setLeftOffset] = useState({left: 0});
+  useEffect(() => {
+    if (popoverContainerRef.current) {
+      let {left} = popoverContainerRef.current.getBoundingClientRect();
+      setLeftOffset({left: -1 * left});
+    }
+  }, []);
```

Comment:
This was taken from the RSP submenu, but I don't think its needed here now that I look at it.

Reply:
This was to position the submenu's container so that it is the full width of the page, used to properly calculate if it needs to flip horizontally. I think you'll still need it?

Reply:
Seems to work without it, so I've removed it for now.